.scripts/run-interface-tests.py

```diff
@@ -0,0 +1,268 @@
+class _Target:
+    """Testable target defined by the arguments to this script.
+
+    See the CLI help for more on the individual arguments.
+    """
+
+    interface_name: str
+    interface_version: str
+    role: str
+    charm_name: str
+    endpoint: str
+
+    def __post_init__(self):
+        self.interface_dir = (
+            _INTERFACES / self.interface_name / 'interface' / self.interface_version
+        )
+        self.interface_tests_path = self.interface_dir / 'tests' / f'test_{self.role}r.py'
+        self.charm_config = self._load_charm_config()
+        self.test_config = self.charm_config.get('test_setup', {})
+
+    def _load_charm_config(self):
+        """Load config for the current interface, role, and charm.
+
+        It is an error if the charm config doesn't exist or is defined more than once.
+        """
+        interface_yaml = yaml.safe_load((self.interface_dir / 'interface.yaml').read_text())
+        charms = interface_yaml[f'{self.role}rs']
+        [charm_config] = [c for c in charms if c['name'] == self.charm_name]
+        logger.info('Charm config: %s', charm_config)
+        return charm_config
+
+
+def _main() -> None:
+    parser = argparse.ArgumentParser()
+    parser.add_argument('name', help='The interface name.')
+    parser.add_argument('version', help='The interface version, e.g. `v2`.')
+    parser.add_argument('role', choices=('provide', 'require'))
+    parser.add_argument('charm_name', help="Used to load the charm's interface test config.")
+    parser.add_argument('endpoint', help='A valid endpoint for the charm + role + interface.')
+    parser.add_argument('--keep', action='store_true', help="Don't delete cloned charm dir.")
+    args = parser.parse_args()
+    target = _Target(
+        interface_name=args.name,
+        interface_version=args.version,
+        role=args.role,
+        charm_name=args.charm_name,
+        endpoint=args.endpoint,
+    )
+    returncode = _interface_tests(target, keep_tempdir=args.keep)
+    sys.exit(returncode)
+
+
+def _interface_tests(target: _Target, keep_tempdir: bool = False) -> int:
+    """Run interface tests for specified interface / version / role / charm / endpoint.
+
+    Executes the following steps:
+        - Reads charm metadata from interface config.
+        - Clones the charm from the specified location.
+        - Writes test files and additional modules to cloned charm repo.
+        - Executes tests with pytest.
+    """
+    # write and execute interface tester file in cloned charm repo
+    with _clone_charm_repo(target.charm_config, keep_tempdir=keep_tempdir) as charm_root:
+        charm_test_file = _write_charm_test_file(target, charm_root=charm_root)
+        # create schema and plugin modules at charm root (included in PYTHONPATH)
+        (charm_root / f'{_SCHEMA_MODULE}.py').symlink_to(target.interface_dir / 'schema.py')
+        (charm_root / f'{_PLUGIN_MODULE}.py').write_text(_PLUGIN_CONTENT)
+        # execute interface tests
```

Comment:
Can we change this comment to what you explained yesterday, that the intention here is to generate `requirements.txt` (or similar)? The actual execution of interface tests is below, so this comment could be moved down.